docs/source/stability_policy.rst

```diff
@@ -0,0 +1,137 @@
+Stability Policy
+================
+
+The ``python-telegram-bot`` library (PTB) tries to provides a stable interface to it’s users.
+As both the user base and the complexity of the library has grown significantly since PTBs conception, we introduced below explicit stability Policy with version NEXT.VERSION.
+
+Large parts of the :mod:`telegram` package are the Python representations of the Telegram Bot API, the stability policy of which PTB can not influence.
+This policy hence includes some special cases for those parts.
+
+What does this policy cover?
+----------------------------
+
+Non-Bot API Functionality
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This policy includes any API or behavior that is documented in this documentation.
+This covers both the :mod:`telegram` package and the :mod:`telegram.ext` package.
+
+Bot API Functionality
+~~~~~~~~~~~~~~~~~~~~~
+
+What doesn’t this policy cover?
+-------------------------------
+
+Introduction of new features or changes of flavors of comparable behavior (e.g. the default for the used HTTP protocol version) are not covered by this policy.
+The internal structure of classes in PTB, i.e. things like the result of ``dir(obj))`` or the contents of ``obj.__dict__``, may change.
+
+Objects are in general not guaranteed to be pickleable (unless stated otherwise) and pickled objects from one version of PTB may not be loadable in future versions.
+We may provide a way to convert pickled objects from one version to another, but this is not guaranteed.
```

Comment:
maybe we can guide them to the pickle transition script if they would like to modify their own?

Reply:
you mean point them in the general direction of transition guides? I don't see the added value of that tbh …

Reply:
yeah, that way they get a starting point on how to migrate their data, but maybe it's a rare case so I'm fine not mentioning it